ui/app/contexts/metametrics.js

```diff
@@ -1,24 +1,101 @@
-import { Component } from 'react'
-import { connect } from 'react-redux'
+import React, { Component, createContext, useEffect, useCallback, useState } from 'react'
+import { connect, useSelector } from 'react-redux'
 import PropTypes from 'prop-types'
-import { withRouter } from 'react-router-dom'
+import { withRouter, useHistory } from 'react-router-dom'
 import { compose } from 'redux'
 import {
   getCurrentNetworkId,
   getSelectedAsset,
   getAccountType,
   getNumberOfAccounts,
   getNumberOfTokens,
+} from '../selectors/selectors'
+import {
   txDataSelector,
-} from '../../../selectors'
-
-import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+} from '../selectors/confirm-transaction'
+import { getEnvironmentType } from '../../../app/scripts/lib/util'
 import {
   sendMetaMetricsEvent,
   sendCountIsTrackable,
-} from '../../utils/metametrics.util'
+} from '../helpers/utils/metametrics.util'
+
+
+export const MetaMetricsContext = createContext(() => {})
+
+export function MetaMetricsProvider ({ children }) {
+  const txData = useSelector(txDataSelector) || {}
+  const network = useSelector(getCurrentNetworkId)
+  const environmentType = getEnvironmentType()
+  const activeCurrency = useSelector(getSelectedAsset)
+  const accountType = useSelector(getAccountType)
+  const confirmTransactionOrigin = txData.origin
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+  const participateInMetaMetrics = useSelector((state) => state.metamask.participateInMetaMetrics)
+  const metaMetricsSendCount = useSelector((state) => state.metamask.metaMetricsSendCount)
+  const numberOfTokens = useSelector(getNumberOfTokens)
+  const numberOfAccounts = useSelector(getNumberOfAccounts)
+  const history = useHistory()
+  const [state, setState] = useState(() => ({
+    currentPath: window.location.href,
+    previousPath: '',
+  }))
+
+  const { previousPath, currentPath } = state
+
+  useEffect(() => {
+    const unlisten = history.listen(() => setState((prevState) => ({
+      currentPath: window.location.href,
+      previousPath: prevState.currentPath,
+    })))
+    // remove this listener if the component is no longer mounted
+    return unlisten
+  }, [history])
+
+  const metricsEvent = useCallback((config = {}, overrides = {}) => {
+    const { eventOpts = {} } = config
+    const { name = '' } = eventOpts
+    const { pathname: overRidePathName = '' } = overrides
+    const isSendFlow = Boolean(name.match(/^send|^confirm/) || overRidePathName.match(/send|confirm/))
+
+    if (participateInMetaMetrics || config.isOptIn) {
+      return sendMetaMetricsEvent({
+        network,
+        environmentType,
+        activeCurrency,
+        accountType,
+        confirmTransactionOrigin,
+        metaMetricsId,
+        numberOfTokens,
+        numberOfAccounts,
+        ...config,
```

Comment:
This appears to be missing `version`. It was added recently - it may have been merged after you started working on this.